### DISCUSSION TITLE
Element output to csv file

### DISCUSSION THREAD

**Discussion by maugis62** [idaholab/moose#26118] [Q&A Modules: Solid mechanics] [answered]
How would I export the maximum principal strain for every element to a csv file?

**Comment by GiudGiud** [comment]
I m not sure but I would start with the output parameters of the TensorMechanics action

Then if that only gets you output to exodus, you can go output to CSV instead using this on the created auxiliary variables
https://mooseframework.inl.gov/source/vectorpostprocessors/ElementValueSampler.html

**Reply by maugis62** [reply, marked as answer]
That worked!  thanks so much!

Just included the following:
```
[VectorPostprocessors]
  [vpp]
    type = ElementValueSampler
    variable = 'max_principal_strain'
    sort_by = id
    execute_on = 'TIMESTEP_END'
  []
[]
```
